packages/core/database/lib/metadata/index.js

```diff
@@ -142,6 +142,11 @@ const createCompoLinkModelMeta = (baseModelMeta) => {
         name: `${baseModelMeta.tableName}_entity_fk`,
         columns: ['entity_id'],
       },
+      {
+        name: `${baseModelMeta.tableName}_unique`,
+        columns: ['entity_id', 'component_id', 'field'],
+        type: 'unique',
+      },
```

Comment:
just curious, is this index related to relations? 
I am trying to understand if it is related to the reordering.

Reply:
It's related to relations yes! It's for 2 things:
- making sure we don't have twice the same relation has it can lead to unexpected results
- be able to use the SQL/knex `onConflict` method: we insert all rows and if it's already existing then we just update the order. It can help for shorter code and maybe performance